src/HexEdit.jl

```diff
@@ -1,57 +1,56 @@
 module HexEdit
 
-using Compat
-import Compat.read
+# using Compat
+# import Compat.read
 
-export HexEd, dump!, edit!,
-       find!
+export HexEd, dump!, edit!, find!
 
-type HexEd # :)
+mutable struct HexEd
     _filesize::Int
     _fh::IO
     _offset::UInt64
-
-    function HexEd(filename)
-        _filesize  = filesize(filename)
-        _fh        = open(filename, "r+")
-        _offset    = 0x00
-        new(_filesize, _fh, _offset)
-    end # constructor HexEd
 end # type HexEd
 
+function HexEd(filename::AbstractString)
```

Comment:
Moving this from an inner constructor to an outer constructor is a change in behavior: `HedEd` now has two constructor methods instead of one. It would be good to reduce the changes here to just those necessary to get the package compatible with 1.0. Changes in behavior can be discussed separately.